skyplane/gateway/gateway_daemon.py

```diff
@@ -56,21 +56,35 @@ def __init__(
             use_compression=use_compression,
             e2ee_key_bytes=e2ee_key_bytes,
         )
-        self.gateway_sender = GatewaySender(
-            region,
-            self.chunk_store,
-            self.error_event,
-            self.error_queue,
-            outgoing_ports=outgoing_ports,
-            use_tls=use_tls,
-            use_compression=use_compression,
-            e2ee_key_bytes=e2ee_key_bytes,
-        )
+        if provider in ("aws", "gcp", "azure", "hdfs"):
+            self.gateway_sender = GatewaySender(
+                region,
+                self.chunk_store,
+                self.error_event,
+                self.error_queue,
+                outgoing_ports=outgoing_ports,
+                use_tls=use_tls,
+                use_compression=use_compression,
+                e2ee_key_bytes=e2ee_key_bytes,
+            )
+        else:
+            self.gateway_sender = GatewaySender(
+                None,
+                self.chunk_store,
+                self.error_event,
+                self.error_queue,
+                outgoing_ports=outgoing_ports,
+                use_tls=use_tls,
+                use_compression=use_compression,
+                e2ee_key_bytes=e2ee_key_bytes,
+            )
```

Comment:
Replace with `region=region if provider in ("aws", "gcp", "azure", "hdfs") else None`

Reply:
Then merge the two blocks

Reply:
Done!